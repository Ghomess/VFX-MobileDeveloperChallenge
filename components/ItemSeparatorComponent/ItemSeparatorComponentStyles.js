import {StyleSheet} from 'react-native';
import {colors} from '../../styles';

export const itemSeparatorComponentStyles = StyleSheet.create({
  separator: {
    borderBottomWidth: 2,

    borderColor: colors.darkgray,
  },
});
