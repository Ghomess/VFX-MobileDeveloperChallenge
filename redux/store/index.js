import {configureStore} from '@reduxjs/toolkit';
import stockSlice from '../reducers/stockSlice';
import pairSlice from '../reducers/pairSlice';
import loadingSlice from '../reducers/loadingSlice';

const store = configureStore({
  reducer: {
    ticker: stockSlice,
    pair: pairSlice,
    loading: loadingSlice,
  },
});

export default store;
