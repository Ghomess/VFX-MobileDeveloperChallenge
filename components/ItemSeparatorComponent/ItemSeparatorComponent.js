import React from 'react';
import {View} from 'react-native';

import {itemSeparatorComponentStyles} from './ItemSeparatorComponentStyles';

export const ItemSeparatorComponent = () => {
  return <View style={itemSeparatorComponentStyles.separator} />;
};
