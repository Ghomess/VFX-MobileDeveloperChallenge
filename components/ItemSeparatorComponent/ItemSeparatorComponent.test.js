import React from 'react';

import {NavigationContainer} from '@react-navigation/native';
import {Provider} from 'react-redux';
import store from '../../redux/store';
import {render, waitFor} from '@testing-library/react-native';
import {ItemSeparatorComponent} from './ItemSeparatorComponent';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    getState: () => ({
      routes: [{name: 'Currency'}],
      index: 0,
    }),
  }),
}));

test('renders correctly', async () => {
  await waitFor(() => {
    const tree = render(
      <Provider store={store}>
        <NavigationContainer>
          <ItemSeparatorComponent />
        </NavigationContainer>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
