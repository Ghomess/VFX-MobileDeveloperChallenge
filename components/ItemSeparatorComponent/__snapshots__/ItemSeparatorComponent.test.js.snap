// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "borderBottomWidth": 2,
      "borderColor": "#424242",
    }
  }
/>
`;
